src/components/Thread/Thread.js

```diff
@@ -35,179 +39,137 @@ const NewThreadText = styled.Text`
-    if (!this.props.thread) {
-      return null;
-    }
-
-    const parentID = this.props.thread.id;
-    const cid = this.props.channel.cid;
-
-    const key = `thread-${parentID}-${cid}`;
-    // We use a wrapper to make sure the key variable is set.
-    // this ensures that if you switch thread the component is recreated
-    return <ThreadInner {...this.props} key={key} />;
-  }
-}
-
-class ThreadInner extends React.PureComponent {
-  static propTypes = {
-    /** Channel is passed via the Channel Context */
-    channel: PropTypes.object.isRequired,
-    /** the thread (just a message) that we're rendering */
-    thread: PropTypes.object.isRequired,
-  };
-
-  constructor(props) {
-    super(props);
-    this.messageList = React.createRef();
-  }
+  useEffect(() => {
+    const loadMoreThreadAsync = async () => {
+      await loadMoreThread();
+    };
 
-  async componentDidMount() {
-    const parentID = this.props.thread.id;
-    if (parentID && this.props.thread.reply_count) {
-      await this.props.loadMoreThread();
+    if (thread?.id && thread?.reply_count) {
+      loadMoreThreadAsync();
     }
-  }
-
-  render() {
-    const {
-      t,
-      thread,
-      additionalParentMessageProps,
-      threadMessages,
-      loadMoreThread,
-      threadHasMore,
-      threadLoadingMore,
-      additionalMessageListProps,
-      autoFocus,
-      additionalMessageInputProps,
-      MessageList: MessageListComponent,
-      MessageInput: MessageInputComponent,
-      disabled,
-    } = this.props;
-    if (!thread) {
-      return null;
-    }
-
-    const read = {};
-    const headerComponent = (
-      <React.Fragment>
-        <Message
-          message={thread}
-          initialMessage
-          threadList
-          readOnly
-          groupStyles={['single']}
-          Message={this.props.Message}
-          // TODO: remove the following line in next release, since we already have additionalParentMessageProps now.
-          {...this.props}
-          {...additionalParentMessageProps}
-        />
-        <NewThread>
-          <NewThreadText>{t('Start of a new thread')}</NewThreadText>
-        </NewThread>
-      </React.Fragment>
-    );
+  }, []);
 
-    return (
-      <React.Fragment>
-        <MessageListComponent
-          messages={threadMessages}
-          HeaderComponent={headerComponent}
-          read={read}
-          threadList
-          loadMore={loadMoreThread}
-          hasMore={threadHasMore}
-          loadingMore={threadLoadingMore}
-          Message={this.props.Message}
-          {...additionalMessageListProps}
-        />
-        <MessageInputComponent
-          parent={thread}
-          focus={autoFocus}
-          disabled={disabled}
-          {...additionalMessageInputProps}
-        />
-      </React.Fragment>
-    );
+  if (!thread) {
+    return null;
   }
-}
 
-export default withTranslationContext(withChannelContext(themed(Thread)));
+  const read = {};
+  const headerComponent = (
```

Comment:
another TODO for eventually moving this to it's own component or nah?

Reply:
Yeah as this is using message it needs a bunch of props from the Thread component at the moment...hopefully in the FC/context changes we can eliminate the need for all those props and move it out nicely.